pipe-cli/mount/pipefuse/archived.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 EPAM Systems, Inc. (https://www.epam.com/)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import collections
+import logging
+from future.utils import iteritems
+
+from pipefuse.fsclient import FileSystemClientDecorator
+
+PATH_SEPARATOR = '/'
+
+
+class ArchivedFilesFilterFileSystemClient(FileSystemClientDecorator):
+
+    def __init__(self, inner, pipe, bucket):
+        """
+        Filters files that have GLACIER or DEEP_ARCHIVE storage class and not restored.
+
+        :param inner: Decorating file system client.
+        :param pipe: Cloud Pipeline API client.
+        :param bucket: Bucket object.
+        """
+        super(ArchivedFilesFilterFileSystemClient, self).__init__(inner)
+        self._inner = inner
+        self._pipe = pipe
+        self._bucket = bucket
+
+    def ls(self, path, depth=1):
+        items = self._inner.ls(path, depth)
+        storage_lifecycle = None
+        result = []
+        for item in items:
+            if not item.is_dir and item.storage_class != 'STANDARD':
+                path = self._normalize_path(path)
+                if storage_lifecycle is None:
+                    storage_lifecycle = self._get_storage_lifecycle(path)
+                file_path = path + item.name
+                if not self._file_restored(file_path, storage_lifecycle):
+                    continue
```

Comment:
It seems that `storage_lifecycle` dictionary can be used more efficiently considering the fact that it can only contain either exact file path or some parent folder path.